src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -285,6 +285,15 @@ private static Binder AdjustBinderForPositionWithinStatement(int position, Binde
                         Debug.Assert(binder is PatternVariableBinder);
                     }
                     break;
+
+                case SyntaxKind.ForEachStatement:
+                    var foreachStmt = (ForEachStatementSyntax)stmt;
+                    if (LookupPosition.IsBetweenTokens(position, foreachStmt.OpenParenToken, foreachStmt.Statement.GetFirstToken()))
+                    {
+                        binder = binder.Next;
```

Comment:
Looks similar to block above. Consider either factoring into shared function or if we could share this logic in another way.